File: Gerenciador de Plano de Aula/Web.Server/Models/Disciplina.cs
using System.ComponentModel.DataAnnotations;

namespace Web.Server.Models
{
    public class Disciplina
    {
        public int Id { get; set; }
        [Required]
        public string Nome { get; set; }
        [Required]
        public int PesoTeoria { get; set; }
        [Required]
        public int PesoPratica { get; set; }
    }
}

File: Sistema de Gestao Academico/Web/ClientApp/src/app/_models/disciplina.ts
export class Disciplina {
  id: number;
  nome: string;
  pesoTeoria: number;
  pesoPratica: number;
  cursoId: number;

  constructor() {
    this.pesoPratica = 0;
    this.pesoTeoria = 0;
  }
}
